File: Anul 2/MAP/Sem9/seminar9/seminar9/domain/StrategyTaskRunner.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace seminar9.domain
{
    class StrategyTaskRunner : ITaskRunner
    {
        private IContainer container;
        public StrategyTaskRunner(Strategy strategy)
        {
           container=new TaskContainerFactory().CreateContainer(strategy);
        }

        public void addTask(Task t)
        {
            container.Add(t);
        }

        public void executeAll()
        {
            while (!container.IsEmpty())
            {
                executeOneTask();
            }
        }

        public void executeOneTask()
        {
            container.Remove().Execute();
        }

        public bool hasTask()
        {
            return container.Size != 0;
        }
    }
}

File: Anul 2/MAP/Seminar 1/src/com/company/sem1/runner/StrategyTaskRunner.java
package com.company.sem1.runner;

import com.company.sem1.container.Container;
import com.company.sem1.container.Strategy;
import com.company.sem1.factory.Factory;
import com.company.sem1.factory.TaskContainerFactory;
import com.company.sem1.model.Task;

public class StrategyTaskRunner  implements TaskRunner{
    private Container container;

    @Override
    public void executeOneTask() {
        if(!container.isEmpty())
            container.remove().execute();
    }

    @Override
    public void executeAll() {
        while(!container.isEmpty())
            executeOneTask();
    }

    @Override
    public void addTask(Task t) {
        container.add(t);
    }

    @Override
    public boolean hasTask() {
        return !container.isEmpty();
    }

    public StrategyTaskRunner(Strategy strategy) {
        this.container = TaskContainerFactory.getInstance().createContainer(strategy);
    }
}
